pkg/services/auth/jwt.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+var SecretKey string
+
+const ExpirationTime = time.Minute * 15
+
+// Claims is a custom JWT claims that contains some token information
+type Claims struct {
+	jwt.StandardClaims
+	Provider      gitproviders.GitProviderName `json:"provider"`
+	ProviderToken string                       `json:"provider_token"`
+}
+
+// Generate generates and signs a new token
+func Generate(secretKey string, expirationTime time.Duration, providerName gitproviders.GitProviderName, providerToken string) (string, error) {
```

Comment:
I think `Generate` it's a bit too generic, I think `GenerateJWT` is probably easier to read and  disambiguate from the provider token